src/control/lib/spdk/src/nvme_control_common.c

```diff
@@ -248,28 +248,48 @@ _discover(prober probe, bool detach, health_getter get_health)
 	return ret;
 }
 
+/* Assign ASCII string as defined by the NVMe spec */
 static int
-copy_ctrlr_data(struct ctrlr_t *cdst, const struct spdk_nvme_ctrlr_data *cdata)
+copy_ascii(char *dst, const void *buf, size_t buf_sz)
 {
-	int	written;
+	const uint8_t	*str = buf;
+	int		 i = 0;
 
-	written = snprintf(cdst->model, sizeof(cdst->model), "%-20.20s",
-			   cdata->mn);
-	if (written >= sizeof(cdst->model))
+	if (buf_sz >= BUFLEN)
 		return -NVMEC_ERR_CHK_SIZE;
 
-	written = snprintf(cdst->serial, sizeof(cdst->serial), "%-20.20s",
-			   cdata->sn);
-	if (written >= sizeof(cdst->serial))
-		return -NVMEC_ERR_CHK_SIZE;
+	/* Trim trailing spaces */
```

Comment:
isspace() could be used if there are other trailing characters like '\t' or '\n' to be trimmed.

Reply:
I'm following the recommendations made in SPDK identity example here as it will be hopefully aligned with what is expected in the NVMe spec.

Reply:
I don't see any harm in using isspace() in this case. We probably don't want to print any of those other whitespace characters either, if they did show up here.

Reply:
Out of curiosity, I went and looked at the NVMe spec, where it defines the convention for string parameters:

> 
> Some parameters are defined as an ASCII string. ASCII strings shall contain only code values 20h through
> 7Eh. For the string “Copyright”, the character “C” is the first byte, the character “o” is the second byte, etc.
> The string is left justified and shall be padded with spaces (ASCII character 20h) to the right if necessary.
> A hexadecimal ASCII string is an ASCII string that uses a subset of the code values: “0” to “9”, “A” to “F”
> uppercase, and “a” to “f” lowercase.
Given that, it seems like you could just stop copying as soon as you hit the first space character.

Reply:
as above, would like to stick with the methodology specified in the SPDK example code unless we have a really good reason to switch from it given that is the framework we are receiving information from